src/components/admin/ChannelConfiguration/ChannelConfiguration.tsx

```diff
@@ -0,0 +1,143 @@
+import React, { useCallback, useEffect, useMemo, useState } from "react";
+import { useAsyncFn } from "react-use";
+import { HeaderButton } from "components/admin/HeaderButton";
+import { Section } from "components/admin/Section";
+import { SectionHeading } from "components/admin/SectionHeading";
+import { SectionTitle } from "components/admin/SectionTitle";
+import { TablePanel } from "components/admin/TablePanel";
+
+import * as adminApi from "api/admin";
+
+import { Channel, MeetingRoomVenue } from "types/venues";
+
+import { WithId } from "utils/id";
+
+import { useShowHide } from "hooks/useShowHide";
+
+import { ConfirmationModal } from "../ConfirmationModal/ConfirmationModal";
+
+import { ChannelAddEditModal } from "./ChannelAddEditModal";
+import { ChannelRow } from "./ChannelRow";
+
+interface ChannelConfigurationProps {
+  space: WithId<MeetingRoomVenue>;
+}
+
+export const ChannelConfiguration: React.FC<ChannelConfigurationProps> = ({
+  space,
+}) => {
+  const {
+    isShown: isShownCreateModal,
+    hide: hideCreateModal,
+    show: showCreateModal,
+  } = useShowHide(false);
+
+  const {
+    isShown: isShownDeleteModal,
+    hide: hideDeleteModal,
+    show: showDeleteModal,
+  } = useShowHide(false);
+
+  const [editingIndex, setEditingIndex] = useState<number>();
+
+  const [isDeleting, setIsDeleting] = useState(false);
+
+  useEffect(() => {
+    // The cleanup below is only triggered when a deletion is complete and
+    // the number of channels is reduced because we have received new data
+    // The use of channels.length in the dependencies is important.
```

Comment:
I understand the idea and smart implementation but why can't we just add the code after the request is complete?